apps/browser/qml/pages/components/PopUpMenuItem.qml

```diff
@@ -0,0 +1,215 @@
+import org.nemomobile.dbus 2.0
+import "." as Browser
+import "../../shared" as Shared
+
+Item {
+    id: root
+    property int horizontalOffset
+    property int iconWidth
+    readonly property int verticalPadding: 3 * Theme.paddingSmall
+
+    height: content.height + verticalPadding * 2
+
+    Column {
+        id: content
+        width: parent.width
+        spacing: Theme.paddingLarge
+        y: verticalPadding
+
+        Column {
+            width: parent.width
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-tab-new"
+                //% "New tab"
+                text: qsTrId("sailfish_browser-la-new_tab")
+                onClicked: {
+                    webView.privateMode = false
+                    overlay.toolBar.enterNewTabUrl()
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-incognito-new"
+                //% "New private tab"
+                text: qsTrId("sailfish_browser-la-new_private_tab")
+                onClicked: {
+                    webView.privateMode = true
+                    overlay.toolBar.enterNewTabUrl()
+                }
+            }
+        }
+
+        Column {
+            width: parent.width
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-search-on-page"
+                enabled: webView.contentItem
+                //% "Search on page"
+                text: qsTrId("sailfish_browser-la-search_on_page")
+
+                onClicked: {
+                    overlay.toolBar.findInPageActive = true
+                    overlay.toolBar.findInPage()
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-add-to-grid"
+                //% "Add to apps grid"
+                text: qsTrId("sailfish_browser-la-add_to_apps_grid")
+                onClicked: pageStack.animatorPush("AddToAppGridDialog.qml",
+                                                  {
+                                                      "url": url,
+                                                      "title": title,
+                                                      "desktopBookmarkWriter": desktopBookmarkWriter,
+                                                      "bookmarkWriterParent": pageStack
+                                                  })
```

Comment:
On pushing the desktop icon page, the menu should also be closed to match the behaviour of other options (i.e. so it acts in the same way as Bookmarks, History and Settings),

Reply:
done

Reply:
Thanks, the transitions all look great now.